packages/manager/modules/observability/src/hooks/useObservabilityFeatures.ts

```diff
@@ -0,0 +1,102 @@
+import { useLayoutEffect, useMemo, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { createError, createSuccess, isSuccess } from '../types/Result';
+import { getSafeDefaultFeature } from '../utils/ObsFeaturesUtils';
+import {
+  FeatureSwitcherItem,
+  ObsFeatureResult,
+  ObsFeatureType,
+} from '../types';
+
+interface UseObservabilityFeaturesResult {
+  currentObsFeature: ObsFeatureType | null;
+  currentObsFeatureResult: ObsFeatureResult;
+  availableFeatures: ObsFeatureType[];
+  hasValidFeature: boolean;
+  shouldShowFeatureSwitcher: boolean;
+  featureSwitcherItems: Array<FeatureSwitcherItem>;
+}
+
+/**
+ * Hook for managing observability features and feature switching functionality.
+ *
+ * This hook handles the logic for determining which observability feature is currently active,
+ * managing feature availability, and providing data for feature switcher UI components.
+ * It automatically syncs the current feature with the URL path and provides fallback
+ * to default features when needed.
+ *
+ * @param enabledFeatures - Array of observability features that are enabled in the module configuration
+ * @param defaultActiveFeature - Optional default feature to use when no feature is explicitly specified
+ * @param productId - Product ID retrieved from URL parameters
+ * @returns Object containing the current feature state, available features, and UI helper components
+ */
+export const useObservabilityFeatures = (
+  enabledFeatures: ObsFeatureType[],
+  defaultActiveFeature?: ObsFeatureType,
+  productId?: string,
+): UseObservabilityFeaturesResult => {
+  const location = useLocation();
+
+  const availableFeatures: ObsFeatureType[] = useMemo(() => {
+    return Object.values(ObsFeatureType).filter((feature) =>
+      enabledFeatures.includes(feature),
+    );
+  }, [enabledFeatures]);
```

Comment:
I don't think useMemo is really useful here.


```suggestion
  const availableFeatures: ObsFeatureType[] = Object.values(ObsFeatureType).filter((feature) =>
      enabledFeatures.includes(feature),
    );
```